### PR TITLE
Add exposed status

### PR DESCRIPTION
**Pull request by huwshimi**
Fixed exposed service style.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a service
- deploy
- expose the service
- the service token should get a thicker border

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3514/
Test PASSed.

**Comment by hatched** [comment]
There is a card to make the expose toggle visible for ghost services as well, could you add that to this branch?

Also I'm wondering if the thicker border shouldn't be blue until committed? Thoughts? As it's not exposed until you actually commit it...

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3519/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3520/
Test FAILed.

**Comment by huwshimi** [comment]
Additional QA:
- add a service
- expose should now be visible
- expose the service
- you should get an entry in the deploy changes
- unexpose the service
- the entry should be removed
- expose the service and commit 
- the service should be committed and remain exposed

Also follow these QA instructions: https://github.com/juju/juju-gui/pull/967#issuecomment-154566533

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3521/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3522/
Test PASSed.

**Comment by hatched** [comment]
Thanks for these updates! :+1: QA :ok:

I was wondering though if we should have the expose icon from the inspector in the commit summary?

**Comment by huwshimi** [comment]
@hatched we could, we'd just need a new unexpose icon to go with it.

File: jujugui/static/gui/src/app/assets/css/canvas/_service.scss [hunk 1]
```diff
@@ -0,0 +1,211 @@
+g.service {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/canvas/_service.scss]: Nice work! thanks for moving.

File: jujugui/static/gui/src/app/assets/svgs/exposed_16.svg [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <path
+         style="color:#000000;display:inline;overflow:visible;visibility:visible;opacity:1;fill:#808080;fill-opacity:1;fill-rule:nonzero;stroke:none;stroke-width:64;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-dashoffset:0;stroke-opacity:1;marker:none;enable-background:accumulate"
+         d="m 240,1384.6699 c -0.23819,0 -0.4761,0.017 -0.71289,0.043 L 239,1385 l -0.27148,-0.2168 c -2.21397,0.3907 -4.12616,1.7759 -5.1875,3.7578 L 235,1390 h 1 l 2,2 v -2 l 1,-1 h 1.99999 l 2,-2 v -2 c -0.7244,-0.2342 -2.23867,-0.3288 -2.99999,-0.3301 z M 238,1392 v 3 l 1,1 v 2 1 c 0.51741,0.1161 0.46973,0.328 1,0.3301 0.24933,0 0.49835,-0.019 0.74609,-0.047 l 0.2539,-0.2832 v -1 l 3,-3 1e-5,-1 -3,-2 z"
+         transform="matrix(1.5,0,0,1.5,-235.99999,-1367.6378)"
+         id="path4243"
+         inkscape:connector-curvature="0"
+         sodipodi:nodetypes="ccccccccccccccccccccccccccc" />
+    </g>
+  </g>
+</svg>
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/svgs/exposed_16.svg]: This is an optimised version: 
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 4.2333345 4.2333334"><g color="#000"><path overflow="visible" fill="none" d="M0 0h4.233v4.233H0z"/><path style="line-height:normal;font-variant-ligatures:normal;font-variant-position:normal;font-variant-caps:normal;font-variant-numeric:normal;font-variant-alternates:normal;font-feature-settings:normal;text-indent:0;text-align:start;text-decoration-line:none;text-decoration-style:solid;text-decoration-color:#000000;text-transform:none;shape-padding:0;isolation:auto;mix-blend-mode:normal" d="M2.117 0C.95 0 0 .95 0 2.117c0 1.166.95 2.116 2.117 2.116s2.116-.95 2.116-2.116C4.233.95 3.283 0 2.117 0zm0 .355c.975 0 1.76.786 1.76 1.762 0 .975-.785 1.762-1.76 1.762C1.14 3.88.355 3.09.355 2.116c0-.976.786-1.762 1.762-1.762z" font-family="sans-serif" white-space="normal" overflow="visible" solid-color="#000000" fill="gray"/><path d="M2.117.177c-.063 0-.126.005-.19.012l-.075.075L1.78.207C1.194.31.69.677.408 1.2l.386.388h.264l.53.53v-.53l.264-.265h.53l.528-.53V.266C2.72.203 2.318.178 2.117.177zm-.53 1.94v.793l.265.265v.794c.137.03.124.086.265.086.066 0 .132-.005.197-.012l.067-.075v-.266l.795-.794v-.264l-.794-.53z" overflow="visible" fill="gray"/></g></svg>

**Comment by anthonydillon** [comment]
Great stuff @huwshimi. :+1: with an optimised expose SVG. Also checked with Luca that we can use the expose icon for both exposing and unexposing in the commit summary as its the same type of action.

**Comment by hatched** [comment]
I just confirmed we can use this icon for both expose and unexpose in the deployment summary

File: jujugui/static/gui/src/app/assets/svgs/exposed_16.svg (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/svgs/exposed_16.svg]: Our new svg sprite system actually optimises these svgs when it builds the sprite, so no need to optimise initially!

**Comment by huwshimi** [comment]
Thanks @anthonydillon and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3540/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3541/
Test PASSed.

**Pull request merged**